### ISSUE TITLE
Error while compiling futures-timer

### ISSUE THREAD

**Issue by ibaryshnikov** [rustasync/tide#305]
```
error[E0277]: the trait bound `ext::TimeoutStream<S>: futures_core::stream::Stream` is not satisfied
   --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-timer-0.2.1/src/ext.rs:170:9
    |
170 | impl<S> TryStream for TimeoutStream<S>
    |         ^^^^^^^^^ the trait `futures_core::stream::Stream` is not implemented for `ext::TimeoutStream<S>`
    |
    = note: required because of the requirements on the impl of `futures_core::stream::private_try_stream::Sealed` for `ext::TimeoutStream<S>`
```
rustc 1.38.0-nightly (00ee1b47f 2019-08-11)

**Comment by taiki-e** [comment]
This is fixed in https://github.com/rustasync/futures-timer/pull/23. It seems necessary to bump futures-timer version.

**Comment by ibaryshnikov** [comment]
Tide doesn't depend directly on `futures-timer`, in addition `cargo tree` produces
```
├── runtime v0.3.0-alpha.6
│   ├── futures-preview v0.3.0-alpha.18 (*)
│   ├── runtime-attributes v0.3.0-alpha.5
│   │   ├── proc-macro2 v0.4.30 (*)
│   │   ├── quote v0.6.13 (*)
│   │   └── syn v0.15.44 (*)
│   ├── runtime-native v0.3.0-alpha.5
│   │   ├── async-datagram v2.2.0
│   │   ├── futures-preview v0.3.0-alpha.18 (*)
│   │   ├── futures-timer v0.2.1
```

**Comment by taiki-e** [comment]
Oops, I missed it.
It seems that [futures-timer has already been updated in the master of runtime](https://github.com/rustasync/runtime/pull/76), so I think that it can be fixed by temporarily adding the following block to Cargo.toml (this must be removed before release).
```toml
[patch.crates-io]
runtime = { git = "https://github.com/rustasync/runtime" }
```

**Comment by ibaryshnikov** [comment]
After doing this I got
```
warning: Patch `runtime v0.3.0-alpha.5 (https://github.com/rustasync/runtime#19a1a825)` was not used in the crate graph.
```
Published `runtime` version is `0.3.0-alpha.6`, but in master it's still `0.3.0-alpha.5`
https://github.com/rustasync/runtime/blob/master/Cargo.toml#L4
And tide depend on `-alpha.6`
https://github.com/rustasync/tide/blob/master/Cargo.toml#L59
